### PR TITLE
Update doc of shouldUseCDIPopulator to reflect changes

### PR DESCRIPTION
**Pull request by akalenyu**
**What this PR does / why we need it**:
I find myself going back a lot to this great function and I noticed
some of the fallback reasons are outdated following:
https://github.com/kubevirt/containerized-data-importer/pull/2765
https://github.com/kubevirt/containerized-data-importer/pull/2873

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/datavolume/controller-base.go [hunk 1]
```diff
@@ -1164,8 +1164,6 @@ func (r *ReconcilerBase) handlePvcCreation(log logr.Logger, syncState *dvSyncSta
 // shouldUseCDIPopulator returns if the population of the PVC should be done using
 // CDI populators.
 // Currently it will use populators only if:
```
Comment (alromeros) [inline pkg/controller/datavolume/controller-base.go:1166]: Maybe rephrase this too since now there's only prerequisite to use populators. I think it's more accurate to say something like `It will use populators if a storage class with CSI Drivers exist`.

**Review by alromeros — COMMENTED** [review]
Cool, thanks for noticing this! Just a couple of suggestions.

File: doc/cdi-populators.md [hunk 2]
```diff
@@ -127,3 +127,9 @@ The behavior will be the same as always with the following key differences:
 > NOTE: Datavolumes and the PVCs they create will be marked with "usePopulator" Annotation to indicate the population is done via CDI populators
 
 For more information of using datavolumes for population check the [datavolume doc](datavolumes.md)
+
+### Fallback to legacy population
+
+In some cases, CDI will fall back to legacy population methods, and thus skip using volume populators:
```
[the remark this answers is not in the document]
Comment (alromeros) [reply doc/cdi-populators.md:133]: I mean, either in `thus skip using volume populators when:` or in the bulletpoints below.

File: pkg/controller/datavolume/controller-base.go (hunk 1 above)
Comment (akalenyu) [reply pkg/controller/datavolume/controller-base.go:1167]: I added a second condition about the usePopulator ann, hopefully that is ok

File: doc/cdi-populators.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (akalenyu) [reply doc/cdi-populators.md]: Added
